### ISSUE TITLE
What's the best way to keep a file specified in the defaults.yml

### ISSUE THREAD

**Issue by jamesprior** [raszi/cleanup_vendor#18]
I'd like to keep `vendor/bundle/ruby/2.7.0/gems/valid_email2-3.4.0/config/disposable_email_domains.txt` since that's a file that the gem actually uses to track disposable email domains at runtime.  It looks like the configuration is additive and there isn't a way to replace it or to specify a different config yaml file.  Am I reading that right?  What do you think would be the most compatible way to make sure that file persists?

**Comment by raszi** [comment]
I will take a look and figure something out for this use case.

**Comment by raszi** [comment]
The command line options are overriding the default values, therefore the following command could work:

```bash
$ cleanup_vendor \
  -f "**/.DS_Store" \
  -f "**/*.{c,cpp,h,hpp,java,log,markdown,md,mk,o,orig,rdoc}" \
  -f "**/gem_make.out"
```

Of course I agree that it is not too convenient, so adding and exclusion option would be better.

The other option is to use the `--dry-run` and the `-0` option together and filter out the unwanted results with `grep -z -v`.
Something like this:

```bash
$ cleanup_vendor --dry-run -0 | grep -z -v 'disposable_email_domains.txt' | xargs -0 rm -r -i
```

**Comment by jamesprior** [comment]
> The command line options are overriding the default values, therefore the following command could work:

Ah yes, I see now - sorry I saw the DEFAULTS.merge and didn't think too hard on it.  CLI options properly override the default options.

> The other option is to use the `--dry-run` and the `-0` option together and filter out the unwanted results with `grep -z -v`.
> Something like this:
> 
> ```shell
> $ cleanup_vendor --dry-run -0 | grep -z -v 'disposable_email_domains.txt' | xargs -0 rm -r -i
> ```

Interesting and a good combination of options!  Thank you!

**Comment by synth** [comment]
I second the need for more control over the defaults.yml. For me, I ran into an issue with the `password_strength` gem, which keeps a `common.txt` file of known bad passwords. I tried the approach above with `--dry-run -0` and it outputs all the files as one blob of text without space separation. 

I can see in the code that you put a NULL character. 
https://github.com/raszi/cleanup_vendor/blob/806e6db4b39d6abaee25b314421bedb4f5fcd0c4/lib/cleanup_vendor.rb#L66

This makes grepping difficult. I assume that is maybe what the `-z` flag is for, but that flag doesn't work with my grep (OSX - posix grep) Is it a POSIX compliant grep flag?

Is there a reason to NULL separate rather than SPACE separate? Or at least make this configurable. NOTE: this didn't work for me on both OSX and on Alpine Docker Linux.

I'm very grateful for the gem, but struggling to get it to work for my case where it needs a bit of customization. I'll try forking the gem and adding some of needed changes and will do a PR for your consideration.
